src/libsystemd/sd-device/device-monitor.c

```diff
@@ -691,7 +691,7 @@ int device_monitor_send_device(
          * If we send to a multicast group, we will get
          * ECONNREFUSED, which is expected.
          */
-        smsg.msg_name = destination ? &destination->snl : &default_destination;
+        smsg.msg_name = (struct sockaddr_nl*) &(destination ?: &default_destination)->nl;
```

Comment:
hmm, I think it would make more sense to use the `->sa` field nstead of the `->nl` field. After all sendmsg() is this generic transport agnostic thing supposedly